pennylane/qchem/observable_hf.py

```diff
@@ -81,12 +81,13 @@ def fermionic_observable(constant, one=None, two=None, cutoff=1.0e-12):
         coeffs = qml.math.concatenate((coeffs, coeffs_two))
         operators = operators + operators_two
 
-    indices_sort = [operators.index(i) for i in sorted(operators)]
+    operators_sort = sorted(operators)
+    indices_sort = [operators.index(i) for i in operators_sort]
```

Comment:
Were we previously accessing the indices from non-sorted operators and that was causing the issues? 🤔

Reply:
We were still sorting the fermionic operators, here I have just defined a variable to avoid sorting the fermionic operator a few times. 

Sorting the fermionic operator is an option but that might lead to nontrivial orders that could be confusing for users that are familiar with other libraries. Instead, I decided to sort the wires at the qubit level.